### PR TITLE
Simplified constructors, added register method, moved tasks

### PR DESCRIPTION
**Pull request by Mooy1**
## Changes
title

## Related Issues
x

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 A Slimefun addon which adds galaxies, star systems, planets, moons, rockets, and spacesuits. It has a large api for addon devs.
 
-Inspiration taken from ClayTech.
\ No newline at end of file
+Inspiration taken from ClayTech and GalactiCraft.
```
Comment (Seggan) [inline README.md]: Perhaps "the mod GalactiCraft"? some people might not know that galacticraft is a mod

File: src/main/java/io/github/addoncommunity/galactifun/Galactifun.java [hunk 2]
```diff
@@ -54,24 +52,11 @@ public void onEnable() {
                 "",
                 "Galactifun is open source, you can contribute or report bugs at: ",
                 getBugTrackerURL(),
-                "Join the Slimefun Addon Community Discord: Discord.gg/V2cJR9ADFU",
+                "Join the Slimefun Addon Community Discord: Discord.gg/", // todo add invite
                 "",
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/Galactifun.java:56]: https://discord.gg/SqD3gg5SAU

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java [hunk 3]
```diff
@@ -18,20 +19,40 @@
...
+    
+    @Nonnull
+    public static DayCycle hours(int hours) {
+        return new DayCycle(hours / 24, hours % 24);
+    }
+    
+    @Nonnull
+    public static DayCycle of(int days, int hours) {
+        return new DayCycle(days + hours / 24, hours % 24);
     }
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java:45]: how is "of" even a method name

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/world/CelestialWorld.java [hunk 4]
```diff
@@ -34,22 +33,21 @@ public static CelestialWorld getByWorld(@Nonnull World world) {
...
-        super(name, orbit, type, choice, celestialBodies);
-        
-        // register the world after subclass loads
-        PluginUtils.runSync(() -> {
-            World world = getWorld();
-            if (world != null) {
-                WORLDS.put(world, this);
-            }
-        });
+    public CelestialWorld(@Nonnull String name, @Nonnull Orbit orbit, @Nonnull CelestialType type, @Nonnull ItemChoice choice) {
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/world/CelestialWorld.java:37]: Why is there an empty line? (kinda nitpicking)

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java (hunk 3 above)
Comment (Seggan) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java:45]: its used a lot in the jdk. i.e `EnumSet.of`

Comment (Seggan) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java:45]: https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html

Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java:45]: i just didnt want daysHours(int days, int hours)

Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/DayCycle.java:45]: maybe the other 2 could be 'ofDays' and 'ofHours'

**Review by Seggan — APPROVED** [review]

**Review by Seggan — APPROVED** [review]

**Comment by Seggan** [comment]
Eh, too lazy to re-review and re-approve. Consider approved from me

File: src/main/java/io/github/addoncommunity/galactifun/base/aliens/Leech.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            if (item != null) {
+                this.eaten.put(entity.getUniqueId(), item);
+            }
+        }
+    }
+
+    @Override
+    public String save(@Nonnull LivingEntity entity) {
+        Set<ItemStack> eaten = this.eaten.get(entity.getUniqueId());
+        if (eaten == null) {
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/aliens/Leech.java]: `@Nullable` or `@Nonnull`, depending on which one is in the interface

File: src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/earth/Earth.java [hunk 6]
```diff
@@ -28,7 +27,7 @@
     
     @Nonnull
     private static World getMainWorld() {
-        World world = Bukkit.getWorld(ConfigUtils.getString("worlds.earth-name"));
+        World world = Bukkit.getWorld(ConfigUtils.getString("worlds.earth-name", "world"));
         if (world == null) {
             throw new IllegalStateException("Failed to read earth world name from config, no default world found!");
         } else {
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/earth/Earth.java]: semicolon not comma

**Review by Seggan — CHANGES_REQUESTED** [review]

**Comment by Mooy1** [comment]
I still gotta test most of the recent commits btw

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java [hunk 7]
```diff
@@ -14,37 +16,48 @@
...
     @Getter 
     private final World.Environment environment;
     @Nonnull
     @Getter
     private final AtmosphericEffect[] effects;
-    
+    @Nonnull
+    @Getter
+    private final Map<AtmosphericGas, Double> composition = new EnumMap<>(AtmosphericGas.class);
+    @Getter
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/Atmosphere.java]: :NotLikeThis: the entire *reason* i did bigdecimal was to have accuracy when filling in the "other gases". if we have a validation error dont blame me

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/AtmosphereBuilder.java [hunk 8]
```diff
@@ -31,7 +33,7 @@ public AtmosphereBuilder setNether() {
 
     /**
      * Note that adding this method <b>will</b> spawn the {@link EnderDragon}
-     *
+     * // TODO see about canceling ender dragon spawn 
      * @return this object
      */
     public AtmosphereBuilder setEnd() {
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/AtmosphereBuilder.java:39]: I have tried to do that when working with sfwarfare, but the bossbar remains and theres a bedrock pillar that wed need to remove

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/AtmosphereBuilder.java [hunk 9]
```diff
@@ -79,8 +87,21 @@ public AtmosphereBuilder enableFire() {
...
     @Nonnull
     public Atmosphere build() {
-        return new Atmosphere(this.oxygenPercentage, this.carbonDioxidePercentage, this.weatherCycle,
-                this.storming, this.thundering, this.flammable, this.environment, this.effects);
+        double percent = 0;
+        
+        for (Double decimal : this.composition.values()) {
+            percent += decimal;
+        }
+
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/api/universe/attributes/atmosphere/AtmosphereBuilder.java]: u can use `this.composition.values().stream().mapToDouble(Double::doubleValue).sum()` to replace all this

**Review by Seggan — COMMENTED** [review]
Note I'm not requesting changes, just commenting.

**Comment by Mooy1** [comment]
did some testing and it all seems to be working as far as i can tell

**Comment by Seggan** [comment]
Can you *pleeeeeease* merge this now

**Comment by ProfElements** [comment]
KLGTM

**Pull request merged**